service/src/main/java/bio/terra/landingzone/service/iam/SamRethrow.java

```diff
@@ -0,0 +1,60 @@
+package bio.terra.landingzone.service.iam;
+
+import bio.terra.common.sam.exception.SamExceptionFactory;
+
+/**
+ * When we use Sam in flights, we want to let InterruptedException fly and be caught by Stairway.
+ * However, when we use Sam outside of flights, we want to propagate the Interrupted exception.
+ * These static methods perform that propagation.
+ */
+public class SamRethrow {
```

Comment:
I copied this from WSM https://github.com/DataBiosphere/terra-workspace-manager/blob/main/service/src/main/java/bio/terra/workspace/service/iam/SamRethrow.java